service/handler.go

```diff
@@ -115,43 +111,8 @@ func EvaluateRequest(
 		return err
 	}
 
-	pathParams := mux.Vars(req)
-	input, err := core.InputFromRequest(req, userInfo, env.ClientTypeHeader, pathParams, nil)
-	if err != nil {
-		return err
-	}
-
-	regoInput, err := core.CreateRegoQueryInput(logger, input, core.RegoInputOptions{
-		EnableResourcePermissionsMapOptimization: permission.Options.EnableResourcePermissionsMapOptimization,
-	})
-	if err != nil {
-		logger.WithField("error", logrus.Fields{"message": err.Error()}).Error("failed rego query input creation")
-		utils.FailResponseWithCode(w, http.StatusInternalServerError, "RBAC input creation failed", utils.GENERIC_BUSINESS_ERROR_MESSAGE)
-		return err
-	}
-
-	evaluatorOptions := &core.EvaluatorOptions{
-		EnablePrintStatements: env.IsTraceLogLevel(),
-	}
-
-	var evaluatorAllowPolicy *core.OPAEvaluator
-	if !permission.RequestFlow.GenerateQuery {
-		evaluatorAllowPolicy, err = partialResultsEvaluators.GetEvaluatorFromPolicy(requestContext, permission.RequestFlow.PolicyName, regoInput, evaluatorOptions)
-		if err != nil {
-			logger.WithField("error", logrus.Fields{"message": err.Error()}).Error("cannot find policy evaluator")
-			utils.FailResponseWithCode(w, http.StatusInternalServerError, "failed partial evaluator retrieval", utils.GENERIC_BUSINESS_ERROR_MESSAGE)
-			return err
-		}
-	} else {
-		evaluatorAllowPolicy, err = core.CreateQueryEvaluator(requestContext, logger, req, permission.RequestFlow.PolicyName, regoInput, nil, evaluatorOptions)
-		if err != nil {
-			logger.WithField("error", logrus.Fields{"message": err.Error()}).Error("cannot create evaluator")
-			utils.FailResponseWithCode(w, http.StatusForbidden, "RBAC policy evaluator creation failed", utils.NO_PERMISSIONS_ERROR_MESSAGE)
-			return err
-		}
-	}
-
-	_, query, err := evaluatorAllowPolicy.PolicyEvaluation(logger, &permission)
+	rondInput := core.NewRondInput(req, env.ClientTypeHeader, mux.Vars(req))
```

Comment:
will this feature be part of the sdk as well or should developers using the sdk use the core as well?

Reply:
Probably, in the future this function will be exposed by a specific package (e.g. http), which could be different by routers for example